### PR TITLE
feat(website): add filter group pattern page

### PR DESCRIPTION
**Pull request by shleewhite**
Adds a new filter group pattern page.

**Note: filter group won't show up in the sidebar until it is marked as "Production" in airtable.**

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: packages/paste-website/src/pages/patterns/filter-group/index.mdx [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+### Complex table filtering
+
+**Complex table filtering** is for users who are adding more than 5 filter parameters, multi-select filter criteria,
+selecting custom query rules, and/or creating complex if/else logic without code to filter down a large data table.
+
+<Callout>
+  <CalloutTitle>This pattern is coming soon!</CalloutTitle>
+  <CalloutText>
+    We've not created this variant of filter group pattern. If you need this for your current or upcoming work, please
+    consider partnering with us to [contribute it](/introduction/contributing/patterns/).
```
Comment (gloriliale) [inline packages/paste-website/src/pages/patterns/filter-group/index.mdx:250]: Issue: the links in these Callouts aren't working. I think if you want to put links inside of a component in mdx, you have to do it w/ the Anchor component.

**Review by gloriliale — COMMENTED** [review]
Woohoo for filter group finally being on the docs! This looks great. I included a couple of copy suggestions and other small items. One thing that the docs don't address, that I always personally found confusing, is how the Search bar works w/ the filters (in that... it doesn't). Is it worth it to add a sentence about it in the General section?

File: packages/paste-website/src/pages/patterns/filter-group/index.mdx [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+### General
+
+The filter group pattern should be used when a user is presented with a page containing a large amount of data that they could be filtering or searching through.
+
+Which filter parameters and functionality you can surface to the user will be highly dependent on the page use case. This pattern should serve as a guide to start with as you design and build your filter group, but will likely be very unique to your specific product or use case.
+
+Be judicious in which filters you make available for your users to represent the most common methods they would be using to reduce the entire data set. Filters being included in the filter group should almost always represent data being displayed in the page data set.
+
+<Img fluid={props.data.defaultState.childImageSharp.fluid} />
+
```
[the remark this answers is not in the document]
Comment (shleewhite) [reply packages/paste-website/src/pages/patterns/filter-group/index.mdx:173]: I added the same marginBottom that paragraphs have, is that better?

File: packages/paste-website/src/pages/patterns/filter-group/index.mdx [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+<Box marginBottom="space70">
+  <Img fluid={props.data.defaultState.childImageSharp.fluid} />
+</Box>
+
+#### Empty state
+
+When the applied filters and/or search does not return any results, use the [empty state pattern](/patterns/empty-state) to inform the user and provide a method to reset all filter and search criteria.
+
+<Img fluid={props.data.emptyState.childImageSharp.fluid} />
```
Comment (SiTaggart) [inline packages/paste-website/src/pages/patterns/filter-group/index.mdx:213]: Suggestion: I think this image is missing the spacer like the their ones.

**Review by SiTaggart — APPROVED** [review]
Aside the empty state image needing a little spacing love, this looks great

File: packages/paste-website/src/pages/patterns/filter-group/index.mdx [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+
+Be judicious in which filters you make available for your users to represent the most common methods they would be using to reduce the entire data set. Filters being included in the filter group should almost always represent data being displayed in the page data set.
+
+<Box marginBottom="space70">
+  <Img fluid={props.data.defaultState.childImageSharp.fluid} />
+</Box>
+
+### Accessibility
+
+The filter group pattern relies on the accessibility features built into components included into the group.
```
Comment (nkrantz) [inline packages/paste-website/src/pages/patterns/filter-group/index.mdx:176]:
```suggestion
The filter group pattern relies on the accessibility features built into components included in the group.
```

File: packages/paste-website/src/pages/patterns/filter-group/index.mdx [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+**Simple table filtering** should be used when you have a table (or data grid) of data and want to allow users
+to temporarily filter and/or search for a subset of the dataset for viewing or analysis. This variant should be used when
+you have a specific amount of parameters you want to allow users to filter by, and not allow users to add their own custom
+query logic.
+
+The following states/logic should be used in your filter group:
+
+- The default value for all filters should be the “All” option, and when the “Clear all” link style button is clicked it should revert all filter parameters back to this default state. The only filter that should have a different default would be the date / time select, which would instead show the default time range your team designated.
+- The filters would all be applied as a single call when the “Apply” button is clicked, instead of filtering the table each time a single filter parameter value is selected.
+- It is recommended to use the [skeleton loader](/components/skeleton-loader) when loading the results of the filters and/or search.
```
Comment (nkrantz) [inline packages/paste-website/src/pages/patterns/filter-group/index.mdx:197]:
```suggestion
- It is recommended to use the [Skeleton Loader](/components/skeleton-loader) when loading the results of the filters and/or search.
```

File: packages/paste-website/src/pages/patterns/filter-group/index.mdx [hunk 6]
```diff
@@ -0,0 +1,287 @@
...
+- The “Apply” and “Clear all” buttons should be in the disabled state on page load, and would enable when at least one of the filter parameters value is changed from the previous state.
+- When using the search input field, users can submit a search by hitting the “enter” key. If the search input is for live filtering, results should update immediately upon load without hitting enter, while being mindful of performance.
+
+Always consider opportunities to reduce cognitive load by limiting the number of filters presented. For the simple table filter group we recommend including no more than 5 filters. If you need to include more than 5 filters, please refer to the [Complex table filtering](#complex-table-filtering) variation.
```
Comment (nkrantz) [inline packages/paste-website/src/pages/patterns/filter-group/index.mdx:201]:
```suggestion
Always consider opportunities to reduce cognitive load by limiting the number of filters presented. For the simple table filter group we recommend including no more than 5 filters. If you need to include more than 5 filters, please refer to the [complex table filtering](#complex-table-filtering) variation.
```

File: packages/paste-website/src/pages/patterns/filter-group/index.mdx [hunk 7]
```diff
@@ -0,0 +1,287 @@
...
+
+#### Default state
+
+<Box marginBottom="space70">
+  <Img fluid={props.data.defaultState.childImageSharp.fluid} />
+</Box>
+
+#### Empty state
+
+When the applied filters and/or search does not return any results, use the [empty state pattern](/patterns/empty-state) to inform the user and provide a method to reset all filter and search criteria.
```
Comment (nkrantz) [inline packages/paste-website/src/pages/patterns/filter-group/index.mdx:211]:
```suggestion
When the applied filters and/or search does not return any results, use the [Empty State pattern](/patterns/empty-state) to inform the user and provide a method to reset all filter and search criteria.
```

**Review by nkrantz — APPROVED** [review]
Looks good, thanks so much for picking this up! Added a few very minor, non-blocking suggestions (just for capitalization consistency etc). No need to update, it's definitely GoodEnough™️

**Pull request merged**